toolchain/check/import_ref.cpp

```diff
@@ -409,42 +437,154 @@ class ImportRefResolver {
     return context_.inst_blocks().AddCanonical(contents);
   }
 
-  // Gets a local version of an imported generic.
-  auto GetLocalGeneric(SemIR::GenericId generic_id) -> SemIR::GenericId {
+  // Gets an incomplete local version of an imported generic. Most fields are
+  // set in the second pass.
+  auto GetIncompleteLocalGeneric(SemIR::InstId decl_id,
+                                 SemIR::GenericId generic_id)
+      -> SemIR::GenericId {
     if (!generic_id.is_valid()) {
       return SemIR::GenericId::Invalid;
     }
 
-    // TODO: Support importing generics. Note that this comes up in the prelude,
-    // so for now we fall back to producing `Invalid` and treating imported
-    // generics as non-generic.
-    return SemIR::GenericId::Invalid;
+    return context_.generics().Add(
+        {.decl_id = decl_id,
+         .bindings_id = SemIR::InstBlockId::Invalid,
+         .self_specific_id = SemIR::SpecificId::Invalid});
   }
 
-  // Gets a local argument list corresponding to the arguments of an imported
-  // specific.
-  auto GetLocalSpecificArgs(SemIR::SpecificId specific_id)
-      -> llvm::SmallVector<SemIR::InstId> {
-    if (!specific_id.is_valid()) {
-      return {};
+  // Gets a local version of the data associated with a generic.
+  auto GetLocalGenericData(SemIR::GenericId generic_id) -> GenericData {
+    if (!generic_id.is_valid()) {
+      return GenericData();
```

Comment:
FWIW, had you considered using std::optional here and for GetLocalSpecificData? I know we often don't, but kind of wonder if it'd actually be better in these cases, particularly due to the `SmallVector`s

Reply:
I hadn't considered that. Thinking about it now, I agree it might avoid some unnecessary work setting up empty `SmallVectors`, but it would mean we no longer have guaranteed RVO.